homeassistant/components/tplink/light.py

```diff
@@ -238,7 +239,9 @@ def do_update_retry(self, update_state_machine: bool) -> None:
             self._is_available = False
 
         if update_state_machine:
-            self.async_write_ha_state()
+            asyncio.run_coroutine_threadsafe(
+                self.async_write_ha_state(), self.hass.loop
```

Comment:
This is a callback, not a coroutine. Use `self.schedule_update_ha_state` instead.